pkg/apis/core/v1alpha1/types_controllerregistration.go

```diff
@@ -57,15 +58,20 @@ type ControllerRegistrationSpec struct {
 // kind (aws-route53, gcp, auditlog, ...).
 type ControllerResource struct {
 	// Kind is the resource kind, for example "OperatingSystemConfig".
-	Kind string `json:"kind" protobuf:"bytes,1,opt,name=kind"`
+	Kind string `json:"kind" protobuf:"bytes,1,rep,name=kind"`
```

Comment:
`rep` stands for [repeated](https://github.com/golang/protobuf/blob/e9dc0d72c1a35f349f6d6871842199ddaaaf9edc/proto/properties.go#L160) and is rather adequate for arrays if my research is correct. I'm not sure about the proper tag for a required field, though. K8s, for instance, defines an `opt` tag for required fields at some places (see [here](https://github.com/kubernetes/api/blob/master/networking/v1beta1/types.go#L305)) while at other places they don't have one at all (see [here](https://github.com/kubernetes/api/blob/master/networking/v1/types.go#L156)).

@ialidzhikov can you shed more light on this?

Reply:
Yeah, thanks for pointing it out and checking, I reverted this change again. It's anyways unrelated to this PR.